### ISSUE TITLE
stage2的训练不收敛

### ISSUE THREAD

**Issue by drj-123456** [Doubiiu/CodeTalker#54]
Hello! We are running on our own data (the vertex data is not 5023*3, but 169-dimensional). The loss in the first stage decreases normally, but the second stage does not converge at all. What is the reason?

**Comment by Doubiiu** [comment]
Hi, since this kind of VQ-stuff is not easy to converge and sensitive to the dataset/task/representation, hypter-parameters, and network architecture (no. of transformer layers, heads...). You may need to modify them to make it work on a customized dataset.

**Comment by Chenh0332** [comment]
> Hello! We are running on our own data (the vertex data is not 5023*3, but 169-dimensional). The loss in the first stage decreases normally, but the second stage does not converge at all. What is the reason?

Hello, I am also training using my own dataset. After completing stage 1 and stage 2, I would like to use the trained model for facial animation generation, but the original demo.py file is not suitable. I would like to know how you showcase the demo using your own features.

**Comment by Chenh0332** [comment]
Hello, when training CodeTalker in Stage 2, I noticed that after several epochs, the loss does not decrease. I output the predicted facial animation features from the model and found that after this point, the output features are all the same. Additionally, when using the “predict()” function to predict facial animation features during the validation phase, I found that all the output facial animation features are identical. What could be the problem?

**Comment by ZhouYihann** [comment]
I also encountered this problem. The loss in training could not be reduced at all, and the output of each frame of visualization was constant. [sad]

**Comment by sd707589** [comment]
@Chenh0332 @bndiehui @ZhouYihann I am meeting the same problem! Have you solved it yet?

**Comment by 2351548518** [comment]
> [@Chenh0332](https://github.com/Chenh0332) [@bndiehui](https://github.com/bndiehui) [@ZhouYihann](https://github.com/ZhouYihann) I am meeting the same problem! Have you solved it yet?我遇到了同样的问题！你解决了吗？

你好，请问您解决这个问题了吗？
